code/modules/projectiles/updated_projectiles/guns/energy.dm

```diff
@@ -371,3 +371,236 @@
+				current_mag.current_rounds += 1 //refund cost of a standard shot.
+				return ready_in_chamber(user, 1)
+			else if (!overcharge && in_chamber.name == over_ammo.name)
+				//to_chat(user, "<span class='warning'>DEBUG: Chamber ammo replaced with regular shot. Chamber: [in_chamber] </span>")
+				in_chamber = null //clean the chamber of the erroneous round
+				current_mag.current_rounds += OVERCHARGE_AMMO_COST //refund cost of an overcharge shot.
+				return ready_in_chamber(user, 2)
+		return in_chamber //Already set!
+	//Let's check on the active attachable. It loads ammo on the go, so it never chambers anything
+	if(active_attachable)
+		if(active_attachable.current_rounds > 0) //If it's still got ammo and stuff.
+			active_attachable.current_rounds--
+			return create_bullet(active_attachable.ammo)
+		else
+			to_chat(user, "<span class='warning'>[active_attachable] is empty!</span>")
+			to_chat(user, "<span class='notice'>You disable [active_attachable].</span>")
+			playsound(user, active_attachable.activation_sound, 15, 1)
+			active_attachable.activate_attachment(src, null, TRUE)
+	else
+		return ready_in_chamber(user)//We're not using the active attachable, we must use the active mag if there is one.
+
+
+/obj/item/weapon/gun/energy/lasgun/ready_in_chamber(mob/user, switch_modes = FALSE)
+	if(current_mag && current_mag.current_rounds > 0)
+		if(current_mag.current_rounds < ammo_per_shot && overcharge)
+			if(istype(src, /obj/item/weapon/gun/energy/lasgun))
+				var/obj/item/weapon/gun/energy/lasgun/L = src
+				L.toggle_chargemode(user)
+				//to_chat(user, "<span class='warning'>DEBUG: Overcharge mode toggled off due to lack of power.</span>")
+		if(switch_modes) //Let the player know wtf is going on with his ammo count if he switches charge modes with mismatched ammo types in the chamber
+			if(switch_modes == 1)
+				to_chat(user, "<span class='warning'>With a hum, [src]'s capacitors draw additional charge as you switch to overcharge mode. [flags_gun_features & GUN_AMMO_COUNTER && current_mag ? "<B>[max(0,current_mag.current_rounds - ammo_per_shot)]</b>/[current_mag.max_rounds]" : ""]</span>")
+			else
+				to_chat(user, "<span class='warning'>With a whine, [src]'s capacitors discharge back into the battery as you switch from overcharge mode. [flags_gun_features & GUN_AMMO_COUNTER && current_mag ? "<B>[max(0,current_mag.current_rounds - ammo_per_shot)]</b>/[current_mag.max_rounds]" : ""]</span>")
+		in_chamber = create_bullet(ammo)
+		current_mag.current_rounds -= ammo_per_shot //Subtract the round from the mag.
+		update_icon(user)
+		return in_chamber
+
+
+//EMPs will fuck with remaining charge
+/obj/item/weapon/gun/energy/lasgun/emp_act(severity)
+	var/amount = round(current_mag.max_rounds * rand(2,severity) * 0.1)
+	if(current_mag.current_rounds < amount)	return 0
+	current_mag.current_rounds = max(0,current_mag.current_rounds - amount)
+	update_icon()
+	current_mag.update_icon()
+	return ..()
```

Comment:
Is there some reason this needs to come last in this proc? Prefer to use `. = ..()` first instead when you can